src/realm/exec/realm_dump.c

```diff
@@ -10,15 +11,26 @@
 #define do_seek fseek
 #endif
 
-typedef struct _Header {
+typedef struct _FileHeader {
     uint64_t m_top_ref[2]; // 2 * 8 bytes
     // Info-block 8-bytes
     uint8_t m_mnemonic[4];    // "T-DB"
     uint8_t m_file_format[2]; // See `library_file_format`
     uint8_t m_reserved;
     // bit 0 of m_flags is used to select between the two top refs.
     uint8_t m_flags;
-} Header;
+} FileHeader;
+
+typedef struct _NodeHeader {
```

Comment:
Where are we storing the node header? It seems something that we already store in the file? Would it not be make more sense to have all these structures defined in some header?